include/z64camera.h

```diff
@@ -95,18 +95,18 @@
 
 // Camera stateFlags. Variety of generic flags
 #define CAM_STATE_0 (1 << 0) // Must be set for the camera to change settings based on the bg surface
-#define CAM_STATE_1 (1 << 1) // Must be set for Camera_UpdateWater to run
-#define CAM_STATE_2 (1 << 2)
+#define CAM_STATE_CHECK_WATER (1 << 1) // Must be set for Camera_UpdateWater to run
+#define CAM_STATE_2 (1 << 2) //  Must be set for the camera to change settings based on the bg surface
 #define CAM_STATE_3 (1 << 3) // Customizable flag for different functions
-#define CAM_STATE_4 (1 << 4)
-#define CAM_STATE_5 (1 << 5)
-#define CAM_STATE_6 (1 << 6)
+#define CAM_STATE_4 (1 << 4) // Related to respawn mode in player
```

Comment:
not sure about `CAM_STATE_4`, in the (only) if block checking that flag, there is also `func_8083B644` called, which looks crazy, but notably checks for A and C-up presses :shrug:

Reply:
Looks to be more related to the end of a mini cutscene, like opening a door (Special 9, Unique 3), opening a large chest (Demo 3), talking in a prerendered room (KeepOn0), talking to an NPC (KeepOn3), Cam pans around and faces Link (KeepOn4, your favorite). By default, this is always set to on every frame (in update), but then the camera update functions listed above will turn it off at the beginning. Then once that mini cutscene is done and the `animState` reaches the end, then `CAM_STATE_4` is turned back on by the function and that signals to player that the cutscene can end. You also see this state being set closely associated with those big if-blocks checking for any controller inputs in the cam-update functions, which are there to end/escape the cam-update func and move on to the next one.

Reply:
This has been readdressed and documented from scratch, so I'm resolving this more general comment for now but feel free to comment on the new docs.